### PR TITLE
Fix hanging "Prebuild in Progress" page

### PR DESCRIPTION
**Pull request by AlexTugarev**
This PR fixes an issue with the hanging "Prebuild in Progress" page.
This PR does not fix missing image builder logs from this very page.

### How to test
* clone https://gitlab.com/alex736/gitpod-large-image into your account
* create a project for it using the preview environment from this PR
* push a change which modifies a) the Dockerfile and/or b) the init script in the .gitpod.yml file
* quickly switch to the project's page and start a workspace when you see the prebuild being started
* now you should see the "Prebuild in Progress" page
* depending on the time "configured" in the Dockerfile of the repo (modification a) you should eventually see the transition to the following workspace creation pages

Fixes https://github.com/gitpod-io/gitpod/issues/8195 and more.

```release-notes
Fix hanging "Prebuild in Progress" page.
```

### CODE REVIEW

**Comment by roboquat** [comment]
@AlexTugarev: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: components/dashboard/src/start/CreateWorkspace.tsx [hunk 1]
```diff
@@ -461,40 +461,29 @@ interface RunningPrebuildViewProps {
...
-            if (done) {
-                // note: this treats "done" as "available" which is not equivalent.
-                // This works because the backend ignores prebuilds which are not "available", and happily starts a workspace as if there was no prebuild at all.
-                prebuildDoneTriggered = true;
-                props.onPrebuildSucceeded();
-                return true;
-            }
-            return false;
-        };
-        const pollIsPrebuildDone = async () => {
```
Comment (AlexTugarev) [inline components/dashboard/src/start/CreateWorkspace.tsx:485]: this was unused at all 🙈

File: components/dashboard/src/start/CreateWorkspace.tsx [hunk 2]
```diff
@@ -461,40 +461,29 @@ interface RunningPrebuildViewProps {
...
-            clearTimeout(pollTimeout!);
-            await checkIsPrebuildDone();
-            pollTimeout = setTimeout(pollIsPrebuildDone, 10000);
-        };
-
         const disposables = watchHeadlessLogs(
             props.runningPrebuild.instanceID,
             (chunk) => logsEmitter.emit("logs", chunk),
-            checkIsPrebuildDone,
```
Comment (AlexTugarev) [inline components/dashboard/src/start/CreateWorkspace.tsx:494]: this was the only option to proceed, i.e. only if headless logs poller checked for `isPrebuildDone` periodically, we'd a chance to proceed. whenever that poller broke, this was a UX dead-end.

File: components/gitpod-protocol/src/protocol.ts [hunk 3]
```diff
@@ -754,7 +754,9 @@ export interface PrebuiltWorkspace {
 
 export namespace PrebuiltWorkspace {
     export function isDone(pws: PrebuiltWorkspace) {
-        return pws.state === "available" || pws.state === "timeout" || pws.state === "aborted";
+        return (
```
Comment (AlexTugarev) [inline components/gitpod-protocol/src/protocol.ts:757]: though `isDone` seems to be very unused after this PR, it seems to be wrong to have the `failed` state missing in here.

Comment (easyCZ) [reply components/gitpod-protocol/src/protocol.ts:757]: If it's unused, let's remove it. It's always available in git history.

Comment (AlexTugarev) [reply components/gitpod-protocol/src/protocol.ts:757]: It's API though.

Comment (easyCZ) [reply components/gitpod-protocol/src/protocol.ts:757]: > It's API though.

What do you mean?

**Review by geropl — APPROVED** [review]
Nice @AlexTugarev , worked as expected for me! :rocket: 

One downside: Logs did not work for me in one case, though, even after reload.
Meanwhile the prebuild view worked flawlessly... :see_no_evil: 
Bumps the importance [respective issue](https://github.com/gitpod-io/gitpod/issues/9132) I guess!

**Comment by AlexTugarev** [comment]
> One downside: Logs did not work for me in one case, though, even after reload.

@geropl, as mentioned before, image build logs never show up in this. I don't think this is a front-end issue, as I've seen the calls were not successful on server side. There is another issue with image build logs for prebuild which is independent of frontend bits.

**Pull request merged**